morango/utils/sync_utils.py

```diff
@@ -224,232 +227,31 @@ def _queue_into_buffer(transfersession):
         queue_rmc_buffer = """INSERT INTO {outgoing_rmcb}
                             (instance_id, counter, transfer_session_id, model_uuid)
                             SELECT instance_id, counter, '{transfer_session_id}', store_model_id
-                            FROM {record_max_counter} AS rmc LEFT JOIN {outgoing_buffer} AS buffer
-                            WHERE buffer.model_uuid = rmc.store_model_id
-                            AND buffer.transfer_session_id = '{transfer_session_id}'
+                            FROM {record_max_counter} AS rmc
+                            INNER JOIN {outgoing_buffer} AS buffer ON rmc.store_model_id = buffer.model_uuid
```

Comment:
Was there a particular reason to change from a LEFT JOIN to an INNER JOIN? Does it impact behavior at all here?

Reply:
Not really, because the `Where` condition will filter out anyways. I just did it because I **thought** it might be slightly faster because it wouldn't be returning all the records from the left side (yet to be proven). The optimizer might do something under the hood which will return the same query but not sure